apps/docs/components/ui/table/TypographyTable.tsx

```diff
@@ -0,0 +1,168 @@
+"use client";
+
+import React from "react";
+import TypographyPreview from "@/components/preview/TypographyPreview";
+import Code from "@/components/ui/code/Code";
+
+import "./table.css";
+
+interface TypographyTableProps {
+    type: string;
+    data: Record<string, { name: string; value: string }[]>;
+}
+
+type FontProperty = "fontFamily" | "fontSize" | "fontWeight" | "lineHeight";
+
+type Size = "3xl" | "2xl" | "xl" | "lg" | "md" | "sm" | "xs";
+
+type GroupedItems = Record<Size, Record<FontProperty, string>>;
+
+interface TokenData {
+    [category: string]: { name: string; value: string }[];
+}
+
+const transformDataToTokenData = (inputData: Record<string, { name: string; value: string }[]>): TokenData => {
+    const tokenData: TokenData = {};
+
+    for (const propertyKey in inputData) {
+        const items = inputData[propertyKey];
+        if (Array.isArray(items)) {
+            tokenData[propertyKey as FontProperty] = items;
+        }
+    }
+
+    return tokenData;
+};
+
+function groupItemsByPropertiesAndSizes(tokenData: TokenData, itemType: string): GroupedItems {
+    const sizes = ["3xl", "2xl", "xl", "lg", "md", "sm", "xs"];
```

Comment:
those could be extracted outside the function as well. and could be used as typing instead of duplicating the string. 

instead of what we have now : 
```

type FontProperty = "fontFamily" | "fontSize" | "fontWeight" | "lineHeight";
type Size = "3xl" | "2xl" | "xl" | "lg" | "md" | "sm" | "xs";

const sizes = ["3xl", "2xl", "xl", "lg", "md", "sm", "xs"];	   
const properties = ["fontFamily", "fontSize", "fontWeight", "lineHeight"];
```
we could change it to : 
```
const Sizes = ["3xl", "2xl", "xl", "lg", "md", "sm", "xs"];	   
const FontProperties = ["fontFamily", "fontSize", "fontWeight", "lineHeight"];
```
then, when you need the type FontProperty  or Size you had before, you can replace them with `keyof typeof Sizes` and `keyof typeof FontProperties`

Reply:
for instance: 
```
groupedItems[sizeKey] = {} as Record<FontProperty, string>;
```

becomes 
```
groupedItems[sizeKey] = {} as Record<keyof typeof FontProperties, string>;
```